cloudify_kubernetes/tasks.py

```diff
@@ -158,6 +158,60 @@ def _do_resource_status_check(resource_kind, response):
         else:
             ctx.logger.debug('status {0}'.format(status))
 
+    elif resource_kind == 'Deployment':
+        conditions = response['status']['conditions']
+        if conditions:
+            for condition in conditions:
+                if condition['type'] == 'Available':
+                    ctx.logger.debug('Deployment condition is Available')
+
+                elif condition['type'] == 'ReplicaFailure':
+                    raise NonRecoverableError(
+                        'Deployment condition is ReplicaFailure ,'
+                        'reason:{0}, message: {1}'
+                        ''.format(condition['reason'], condition['message']))
+
+                elif condition['type'] == 'Progressing':
+                    raise OperationRetry(
+                        'Deployment condition is Progressing')
+        else:
+            raise OperationRetry('Deployment condition is not ready yet')
+
+    elif resource_kind == 'PersistentVolumeClaim':
+        status = response['status']['phase']
+        if status in ['Pending', 'Available']:
+            raise OperationRetry(
+                'PersistentVolumeClaim status is {},'
+                'No persistent volume is detected'.format(status))
+
+        elif status == 'Bound':
+            ctx.logger.debug('PersistentVolumeClaim status is Bound')
+
+    elif resource_kind == 'PersistentVolume':
+        status = response['status']['phase']
+        if status in ['Bound', 'Available']:
+            ctx.logger.debug('PersistentVolume status is {0}'.format(status))
```

Comment:
Same as above.

Reply:
For both these statuses, the ``PersistentVolume`` will be on ``Bound`` status if there is request mapped to it and if there is no any claim request it will be on  ``Available`` which means it created successfully